src/backends/tensorflow/base_backend.cc

```diff
@@ -437,19 +442,47 @@ ReadStringOutputTensor(
       }
 
       void* content;
+      TRTSERVER_Memory_Type preferred_memory_type = TRTSERVER_MEMORY_CPU;
```

Comment:
All this tensor copy logic (including for string datatype), should be in a common shared area and handle all the possible cases. Should also return the bool indicating if sync is needed. @GuanLuo is this some of the "common code" you were mentioning earlier. @CoderHam Don't need to fix for this change be we do need to do this cleanup so please file a ticket for it.

Reply:
This is not the part that I think of originally. I don't think we need to generalize it yet, since currently only ONNX and TensorFlow backends are accepting string data type and the two backends handle string type differently. But still, we can unify the logic and move it to common area if other frameworks are going to support string data type.

Reply:
Filled a ticket for the cleanup of the tensor copy logic